R/tt_as_df.R

```diff
@@ -116,17 +118,19 @@ as_result_df <- function(tt, spec = NULL,
         )
 
         # number of values difference mask between cellvals and cellvals_init (TRUE if different)
-        dmc <- lengths(unlist(cellvals, recursive = FALSE)) != lengths(unlist(cellvals_init, recursive = FALSE))
+        dmc <- .lengths_with_nulls(unlist(cellvals, recursive = FALSE)) !=
+          .lengths_with_nulls(unlist(cellvals_init, recursive = FALSE))
         dmc <- matrix(dmc, nrow = nrow(cellvals), ncol = ncol(cellvals))
 
         # Mainly used for debugging
-        selected_rows_to_print <- mf_strings(mf_tt)[-seq_len(mf_nlheader(mf_tt)), , drop = FALSE]
-        selected_rows_to_print <- cbind(
-          which(apply(dmc, 1, any, simplify = TRUE)),
-          as.data.frame(selected_rows_to_print[apply(dmc, 1, any), , drop = FALSE])
-        )
-        colnames(selected_rows_to_print) <- c("row_number", "row_name", colnames(cellvals_init))
-        warning_msg <- if (verbose) {
+        warning_msg <- if (verbose) { # beware that \n will break this (use make_row_df(tt)$self_extent for fix)
+          selected_rows_to_print <- mf_strings(mf_tt)[-seq_len(mf_nlheader(mf_tt)), , drop = FALSE]
+          selected_rows_to_print <- selected_rows_to_print[!make_row_df(tt)$node_class == "LabelRow", , drop = FALSE]
+          selected_rows_to_print <- cbind(
+            which(apply(dmc, 1, any, simplify = TRUE)),
+            as.data.frame(selected_rows_to_print[apply(dmc, 1, any), , drop = FALSE])
+          )
+          colnames(selected_rows_to_print) <- c("row_number", "row_name", colnames(cellvals_init))
```

Comment:
i have a feeling we might have to come back with warning messages, it is ok for now.